src/hotspot/share/jfr/writers/jfrStreamWriterHost.inline.hpp

```diff
@@ -77,7 +79,18 @@ inline void StreamWriterHost<Adapter, AP>::write_bytes(const u1* buf, intptr_t l
   while (len > 0) {
     const unsigned int nBytes = len > INT_MAX ? INT_MAX : (unsigned int)len;
     const ssize_t num_written = os::write(_fd, buf, nBytes);
-    guarantee(num_written > 0, "Nothing got written, or os::write() failed");
+
+    if (errno==ENOSPC) {
+        char msg[] = "Failed to write to jfr stream because no space left on device";
+        log_error(jfr, system)("%s", msg);
+        JavaThread* jt = JavaThread::current();
+        ThreadInVMfromNative transition(jt);
+        JfrJavaSupport::abort(JfrJavaSupport::new_string(msg, jt), jt, false);
+    }
+    else {
```

Comment:
The else block can be removed. Just put the guarantee inline with the other code.

Reply:
Thank you so much. You're right. I removed the else block.